Userland/Libraries/LibJIT/X86_64/Assembler.h

```diff
@@ -877,18 +877,34 @@ struct X86_64Assembler {
 
     void native_call(
         u64 callee,
-        Vector<Operand> const& preserved_registers = {},
-        Vector<Operand> const& stack_arguments = {})
+        Vector<Operand> const& arguments,
```

Comment:
Like previously discussed in discord, I'd prefer if native_call only took the arguments which might be passed on the stack. As you can see in the Compiler.cpp change this adds a lot of redundant passing of ARG0, ARG1, ARG2, ...

Reply:
Yes but at the same time removes a lot of redundant `mov(ARGn, expr)`. We could even change some APIs to, instead of doing a `mov` just returning the proper register and offset and then it'd be even cleaner.

Reply:
There's no difference between `mov(ARGn, expr)` and passing expr as the nth argument.
However all of these calls now pass a `VM_POINTER` which is actually redundant compared to before where we didn't have any `mov ARG0, VM_POINTER` for every call.
And in practice we rarely do `mov(ARGn, expr)`, but often do `load_accumulator(ARGn)` and `load_vm_register(ARGn)`.

Reply:
It's not redundant because on Aarch64 and RiscV64 it's impossible to keep VM in ARG0 all the time as return registers overlap with parameter registers on those ABIs. One way or another we will need `mov(ARG0, VM_POINTER)` before every `native_call` anyway